patches/chromium/feat_add_data_transfer_to_requestsingleinstancelock.patch

```diff
@@ -499,23 +524,37 @@ index ec725b44296266bea1a51aea889463a0bba8449c..3bb74c08cd78b11cd9925a6bfafc62d0
  // Code roughly based on Mozilla.
  ProcessSingleton::NotifyResult ProcessSingleton::NotifyOtherProcess() {
    TRACE_EVENT0("startup", "ProcessSingleton::NotifyOtherProcess");
-@@ -290,8 +426,9 @@ ProcessSingleton::NotifyResult ProcessSingleton::NotifyOtherProcess() {
+@@ -290,8 +443,9 @@ ProcessSingleton::NotifyResult ProcessSingleton::NotifyOtherProcess() {
      return PROCESS_NONE;
    }
  
 -  switch (chrome::AttemptToNotifyRunningChrome(remote_window_)) {
 +  switch (chrome::AttemptToNotifyRunningChrome(remote_window_, additional_data_)) {
      case chrome::NOTIFY_SUCCESS:
-+      ReadAck(notification_ack_callback_);
++      ReadAck(notification_ack_callback_, program_name_, user_data_dir_);
        return PROCESS_NOTIFIED;
      case chrome::NOTIFY_FAILED:
        remote_window_ = NULL;
-@@ -429,6 +566,18 @@ bool ProcessSingleton::Create() {
+@@ -429,6 +583,32 @@ bool ProcessSingleton::Create() {
            << "Lock file can not be created! Error code: " << error;
  
        if (lock_file_ != INVALID_HANDLE_VALUE) {
 +        // We are the first instance. Create a pipe to send out ack data.
-+        ack_pipe_ = ::CreateNamedPipe(kPipeName,
++        // Create a per-process pipename using a combination of the
++        // username, process id, thread id, and program name. Pipe names max
++        // at 256 characters, can include any character other than a backslash
++        wchar_t user_name[256];
++        DWORD size = std::size(user_name);
++        if (::GetUserName(user_name, &size) == 0 || size < 1) {
++          LOG(ERROR) << "User name could not be found for pipe name.";
++        }
++        std::string identifier = base::NumberToString(::GetCurrentProcessId()) +
++              base::NumberToString(::GetCurrentThreadId());
++        std::wstring pipe_name = base::UTF8ToWide("\\\\.\\pipe\\" + program_name_ +
++              identifier);
```

Comment:
```suggestion
+        std::wstring pipe_name = base::StringPrintf(
+              L"\\\\.\\pipe\\%s.%lu.%lu", program_name_, ::GetCurrentProcessId(), ::GetCurrentThreadId());
```

Reply:
This change was causing some build errors and caused occasional crashes while I was testing (which was odd!). I looked into it, and I think the issue is using StringPrintf to cast a wide string instead of a string - it's technically allowed as an override on Windows, but there's a comment above the override warning about using it:
```
// Note: Unfortunately compile time checking of the format string for UTF-16
// strings is not supported by any compiler, thus these functions should be used
// carefully and sparingly.
```

I ended up leaving the formatting as is, because it felt a bit more stable - sorry, just didn't want you to think that I disregarded this suggestion! 🙇‍♀️